install/db/dbmigrate_3.5-3.6.php

```diff
@@ -103,6 +103,42 @@ function updateHash($dbManager, $tableName)
   return count($rows);
 }
 
+/**
+ * @brief Update the sha256 column of the table with value from textfinding.
+ * @param DbManager $dbManager
+ * @param string $tableName
+ * @return integer Number of entries updated
+ */
+function updateSHA256($dbManager, $tableName)
+{
+    if($dbManager == null){
+        echo "No connection object passed!\n";
+        return false;
+    }
+    if(DB_TableExists($tableName) != 1) {
+        // Table does not exists (migrating from old version)
+        echo "Table $tableName does not exists, not updating!\n";
+        return 0;
+    }
+    $sql = "SELECT ".$tableName.".".$tableName . "_pk AS id ".
+        "FROM $tableName WHERE $tableName.".$tableName."_sha256 is NULL";
+    $statement = __METHOD__ . ".getNullSHA256.$tableName";
+    $rows = $dbManager->getRows($sql, [], $statement);
+    $sql = "UPDATE $tableName
+            SET ".$tableName."_sha256 = $2
+            WHERE " . $tableName . "_pk = $1;";
+
+    $statement = __METHOD__ . ".updateHashOf.$tableName";
+    $dbManager->prepare($statement, $sql);
+    foreach ($rows as $row) {
+        $dbManager->execute($statement, [
+            $row["id"],
+            hash('sha256',  file_get_contents(RepPath($row['id'],"files")))
```

Comment:
It will be a better option to leave calculation of hash on PHP instead of loading whole file in memory. There might be some files in repository which are more than huge and can not be loaded in memory due to limits in PHP.

Suggested function : https://www.php.net/manual/en/function.hash-file.php

Reply:
Tested in PHP with memory limit set to 12Mb and calculating hash for file of 500Mb+ size:

```php
php > echo hash('sha256', file_get_contents('gcc-4.9.4.tar'));
PHP Fatal error:  Allowed memory size of 12582912 bytes exhausted (tried to allocate 580216864 bytes) in php shell code on line 1
PHP Stack trace:
PHP   1. {main}() php shell code:0
PHP   2. file_get_contents() php shell code:1
php > echo hash_file('sha256', 'gcc-4.9.4.tar');
c75551154048676f11fe2266a7041cdf931f0b959e3a06c43c99f362249c0f8f
```

Hash value matches `sha256sum` output:
```bash
$ sha256sum gcc-4.9.4.tar
c75551154048676f11fe2266a7041cdf931f0b959e3a06c43c99f362249c0f8f  gcc-4.9.4.tar
```